xds/src/main/java/io/grpc/xds/GcpAuthenticationConfig.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import io.grpc.xds.Filter.FilterConfig;
+
+public class GcpAuthenticationConfig implements FilterConfig {
```

Comment:
`final`. Definitely not `public`.

It's typical for us to have this be a `static` nested class of the filter. That keeps it close to its parse method and the only user of it.